DOCS: Added more detail to titleTemplate. (#5057)
nuxt/framework: docs/content/2.guide/2.features/4.head-management.md

::ReadMore{link="/api/composables/use-head"}
::

## Title Templates

> In previous versions of Nuxt, you could use the `titleTemplate` option to dynamically customize the title of your site. However, this is no longer supported in Nuxt 3, and can instead be done via a plugin.

Reviewer: ```suggestion
You can use the `titleTemplate` option to provide a dynamic template for customizing the title of your site, for example, by adding the name of your site to the title of every page.
```

> Create a `title.js` plugin within your `plugins` folder, with the following code:

Reviewer: ```suggestion
The `titleTemplate` can either be a string, where `%s` is replaced with the title, or a function. If you want to use a function (for full control), then this cannot be set in your `nuxt.config`, and it is recommended instead to set it within your `app.vue` file, where it will apply to all pages on your site:
```

> ```js
> export default defineNuxtPlugin(() => {
>   useHead({
>     titleTemplate: (titleChunk) => {
>       return titleChunk ? `${titleChunk} - Site Title` : 'Site Title';
>     },
>   });
> });
> ```
> 
> Then you can set a title as you usually would:

Reviewer: ```suggestion
```

```vue [/pages/my-page.vue]
<script setup>
  useHead({

>     title: 'My Page'

Reviewer: ```suggestion
    titleTemplate: (titleChunk) => {
      return titleChunk ? `${titleChunk} - Site Title` : 'Site Title';
    },
```

Reviewer: the function approach _should_ work everywhere but in `nuxt.config`.

Author: Even with the following example in `app.vue`:
```vue
<script setup>
useHead({
  titleTemplate: (titleChunk) => {
    return titleChunk ? `${titleChunk} - Site` : 'Site: Default Text';
  },
});
</script>

<template>
  <div>
    <NuxtLayout>
      <NuxtPage/>
    </NuxtLayout>
  </div>
</template>
```

If you don't pass a `useHead` to your page, it won't default it.

  });
</script>
```

> This will now appear as 'My Page - Site Title' in the browser tab. You can also pass `null` to default to the site title.

Reviewer: ```suggestion

Now, if you set the title to `My Page` with `useHead` on another page of your site, the title would appear as 'My Page - Site Title' in the browser tab. You can also pass `null` to default to the site title.
```

::ReadMore{link="/api/composables/use-head"}
::

## Title Templates
In previous versions of Nuxt, you could use the `titleTemplate` option to dynamically customize the title of your site. However, this is no longer supported in Nuxt 3, and can instead be done via a plugin.

Create a `title.js` plugin within your `plugins` folder, with the following code:

```js
export default defineNuxtPlugin(() => {
  useHead({
    titleTemplate: (titleChunk) => {
      return titleChunk ? `${titleChunk} - Site Title` : 'Site Title';
    },
  });
});
```

Then you can set a title as you usually would:

> ```js

Reviewer 2: ```suggestion
```vue
```

> ```vue
> // pages/my-page.vue

Reviewer 3: ```suggestion
```vue [/pages/my-page.vue]
```

<script setup>
  useHead({
    title: 'My Page'
  });
</script>
```

> This will now appear as 'My Page - Site Title' in the browser tab, you can also pass `null` to default to the site title.

Reviewer 3: ```suggestion
This will now appear as 'My Page - Site Title' in the browser tab. You can also pass `null` to default to the site title.
```
